iota/caip10.md

```diff
@@ -0,0 +1,210 @@
+to a node to validate the targetted chain or network. Include a sample
+request/response and add the relevant documentation to the `## References`
+section below if possible, as well as an explanation of any steps needed to
+validate the results, calculate checksums, persist session metadata or nonces,
+ etc. -->
+
+Any Account ID, conforming to the aforementioned syntax is a valid IOTA Account ID,
+indipendently of whether it is already in use by a wallet or on-chain object.
+
+It is possible to check whether a given Account ID refers to a key-pair derived address,
+an object, or a smart contract package by invoking the `iota_getObject` JSON-RPC API.
+
+For instance, checking IOTA address `0x53e4567ccafa5f36ce84c80aa8bc9be64e0d5ae796884274aef3005ae6733809` with
+
+```json 
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "method": "iota_getObject",
+  "params": [
+    "0x53e4567ccafa5f36ce84c80aa8bc9be64e0d5ae796884274aef3005ae6733809",
+    {
+      "showType": true,
+      "showOwner": true,
+      "showPreviousTransaction": false,
+      "showDisplay": false,
+      "showContent": false,
+      "showBcs": false,
+      "showStorageRebate": false,
+    }
+  ]
+}
+```
+
+yields the following response:
+
+```json
+{
+  "jsonrpc": "2.0",
+  "result": {
+    "data": {
+      "objectId": "0x53e4567ccafa5f36ce84c80aa8bc9be64e0d5ae796884274aef3005ae6733809",
+      "version": "1",
+      "digest": "33K5ZXJ3RyubvYaHuEnQ1QXmmbhgtrFwp199dnEbL4n7",
+      "type": "0x2::coin::Coin<0x2::iota::IOTA>",
+      "owner": {
+        "AddressOwner": "0xc8ec1d5b84dd6289e193b9f88de4a994358c9f856135236c3e75a925e1c77ac3"
+      }
+    }
+  },
+  "id": 1
+}
+```
+
+which shows that address `0x53e4567ccafa5f36ce84c80aa8bc9be64e0d5ae796884274aef3005ae6733809` is actually
+an IOTA object of type `0x2::coin::Coin<0x2::iota::IOTA>` - i.e. an IOTA Coin - owned by address
+`0xc8ec1d5b84dd6289e193b9f88de4a994358c9f856135236c3e75a925e1c77ac3`.
+
+If we were to do the same for the coin owner's address
+`0xc8ec1d5b84dd6289e193b9f88de4a994358c9f856135236c3e75a925e1c77ac3` we'd get the following response:
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "result": {
+    "error": {
+      "code": "notExists",
+      "object_id": "0xc8ec1d5b84dd6289e193b9f88de4a994358c9f856135236c3e75a925e1c77ac3"
+    }
+  }
+}
+```
+
+Which hints at the fact that address `0xc8ec1d5b84dd6289e193b9f88de4a994358c9f856135236c3e75a925e1c77ac3` is not
+an object, but a key-derived address.
+
+## Rationale
+
+<!-- Explain here how the mapping or translation between native identifiers and
+CAIP-10 identifiers was arrived at, history and pre-history, etc.-->
+
+IOTA addresses are lowercase-hex encoded 32 bytes string, which in the case of key-pair derived addresses - i.e.
+traditional accounts - are computed by hashing with the `blake2b` algorithm a public key's bytes prepended by a
+single tag byte that identifies the type of key used - e.g. `ed25519`, `secp256k1`, etc.
```

Comment:
```suggestion
IOTA address currently supports pure Ed25519, Secp256k1, Secp256r1, and MultiSig with corresponding flag bytes of 0x00 (not prepended), 0x01, 0x02, and 0x03, respectively.
```
This is just my best guess, it's actually unclear from [the docs](https://docs.iota.org/developer/cryptography/transaction-auth/keys-addresses#address-format), whether Ed25519 have a 0x00 byte prepended or no flag-byte prepended... maybe double-check before merging or rewording/refactoring my suggestion! but either way, you get the gist-- however it works, this is the level of detail that's most useful to include in the CASA profile.

Oh, and these docs should definitely be linked in the ## References section!